src/node_main_instance.cc

```diff
@@ -0,0 +1,160 @@
+#include "node_main_instance.h"
+#include "node_internals.h"
+#include "node_options-inl.h"
+#include "node_v8_platform-inl.h"
+
+namespace node {
+
+using v8::Context;
+using v8::HandleScope;
+using v8::Isolate;
+using v8::Local;
+using v8::Locker;
+using v8::SealHandleScope;
+
+NodeMainInstance::NodeMainInstance(uv_loop_t* event_loop,
+                                   const std::vector<std::string>& args,
+                                   const std::vector<std::string>& exec_args)
+    : args_(args),
+      exec_args_(exec_args),
+      array_buffer_allocator_(CreateArrayBufferAllocator()),
```

Comment:
```suggestion
      array_buffer_allocator_(CreateArrayBufferAllocator::Create()),
```

(I’d consider `CreateArrayBufferAllocator()` legacy; also, this allows us to use a plain `unique_ptr`)

Reply:
@joyeecheung Sorry, just noticing that my suggestion was bogus – I should have dropped the `Create` in front of the identifier as well :smile: